### PR TITLE
feat(inputs.vsphere): Flag for more lenient behavior when connect fails on startup

### LINKED ISSUE #10523: [inputs.vsphere] Telegraf exit on init if conn to vsphere fails
### Relevent telegraf.conf

```toml
[global_tags]

[agent]
debug = true
quiet = false
interval = "1m"
round_interval = false
metric_batch_size = 1000
metric_buffer_limit = 100000
collection_jitter = "0s"
flush_interval = "10s"
flush_jitter = "0s"
precision = "s"
hostname = ""
omit_hostname = true

[outputs]
[[outputs.influxdb]]
urls = [ "http://127.0.0.1:8429",]
exclude_database_tag = true
skip_database_creation = true

[[inputs.mem]]

[[inputs.vsphere]]
vcenters = [ "https://localhost/sdk" ]
username = "vmuser"
password = "pswd"
tagexclude = [ "dcname", "source", "vcenter", "uuid", "moid",]
vm_metric_include = [ "cpu.costop.summation", "cpu.ready.summation", "cpu.wait.summation", "cpu.run.summation", "cpu.idle.summation", "cpu.used.summation", "cpu.demand.average", "cpu.usagemhz.average", "cpu.usage.average", "mem.active.average", "mem.granted.average", "mem.consumed.average", "mem.usage.average", "mem.vmmemctl.average", "net.bytesRx.average", "net.bytesTx.average", "net.droppedRx.summation", "net.droppedTx.summation", "net.usage.average", "power.power.average", "virtualDisk.numberReadAveraged.average", "virtualDisk.numberWriteAveraged.average", "virtualDisk.read.average", "virtualDisk.readOIO.latest", "virtualDisk.throughput.usage.average", "virtualDisk.totalReadLatency.average", "virtualDisk.totalWriteLatency.average", "virtualDisk.write.average", "virtualDisk.writeOIO.latest", "sys.uptime.latest",]
host_include = []
host_exclude = [ "*",]
cluster_include = []
cluster_exclude = [ "*",]
datastore_include = []
datastore_exclude = [ "*",]
datacenter_include = []
datacenter_exclude = [ "*",]
separator = "_"
collect_concurrency = 10
discover_concurrency = 10
custom_attribute_include = [ "*",]
custom_attribute_exclude = []
insecure_skip_verify = true
historical_interval = "5m"
```

### Logs from Telegraf

```
Jan 26 08:40:46 centos8 telegraf[15954]: 2022-01-26T08:40:46Z I! Starting Telegraf 1.21.1
Jan 26 08:40:46 centos8 telegraf[15954]: 2022-01-26T08:40:46Z I! Loaded inputs: mem vsphere
Jan 26 08:40:46 centos8 telegraf[15954]: 2022-01-26T08:40:46Z I! Loaded aggregators:
Jan 26 08:40:46 centos8 telegraf[15954]: 2022-01-26T08:40:46Z I! Loaded processors:
Jan 26 08:40:46 centos8 telegraf[15954]: 2022-01-26T08:40:46Z I! Loaded outputs: influxdb
Jan 26 08:40:46 centos8 telegraf[15954]: 2022-01-26T08:40:46Z I! Tags enabled:
Jan 26 08:40:46 centos8 telegraf[15954]: 2022-01-26T08:40:46Z I! [agent] Config: Interval:1m0s, Quiet:false, Hostname:"", Flush Interval:10s
Jan 26 08:40:46 centos8 telegraf[15954]: 2022-01-26T08:40:46Z D! [agent] Initializing plugins
Jan 26 08:40:46 centos8 telegraf[15954]: 2022-01-26T08:40:46Z D! [agent] Connecting outputs
Jan 26 08:40:46 centos8 telegraf[15954]: 2022-01-26T08:40:46Z D! [agent] Attempting connection to [outputs.influxdb]
Jan 26 08:40:46 centos8 telegraf[15954]: 2022-01-26T08:40:46Z D! [agent] Successfully connected to outputs.influxdb
Jan 26 08:40:46 centos8 telegraf[15954]: 2022-01-26T08:40:46Z D! [agent] Starting service inputs
Jan 26 08:40:46 centos8 telegraf[15954]: 2022-01-26T08:40:46Z I! [inputs.vsphere] Starting plugin
Jan 26 08:40:46 centos8 telegraf[15954]: 2022-01-26T08:40:46Z D! [inputs.vsphere] Creating client: localhost
Jan 26 08:40:46 centos8 telegraf[15954]: 2022-01-26T08:40:46Z E! [telegraf] Error running agent: starting input inputs.vsphere: Post "https://localhost/sdk": dial tcp 127.0.0.1:443: connect: connection refused
Jan 26 08:40:46 centos8 systemd[1]: telegraf.service: Main process exited, code=exited, status=1/FAILURE
Jan 26 08:40:46 centos8 systemd[1]: telegraf.service: Failed with result 'exit-code'.
```

### System info

centos 8

### Docker

_No response_

### Steps to reproduce

1. set invalid vcenters url
2. start telegraf

### Expected behavior

At least a log message to STDERR, on init, about a failing conn to a vcenter server (with no retry mechanism)

### Actual behavior

telegraf.service: Main process exited

### Additional info

same issue, but mongodb plugin
https://github.com/influxdata/telegraf/issues/10078

### PR DESCRIPTION
**Pull request by prydin**
- [x] Updated associated README.md.
- [x] Wrote appropriate unit tests.
- [x] Pull request title or commits are in [conventional commit format]

resolves #10523

Added the server_disconnected_behavior flag mimicking the behavior of the MongoDB-plugin. A value of "error" (default) exits telegraf if the connection to vCenter failed on startup. A value of "ignore" prints a warning and continues to try to connect every collection cycle.

### CODE REVIEW

**Comment by prydin** [comment]
The merge history looks a bit stupid, but I've checked it and it's fine. Just an artifact of how I brought the changes I made earlier today into this branch. Yes, I should have rebased instead... :)

**Review by powersj — APPROVED** [review]
Thanks

@srebhan when you review, the leaf changes in finder.go and directory readme changes were from a previous PR merged thursday morning

File: plugins/inputs/vsphere/endpoint.go [hunk 1]
```diff
@@ -304,7 +304,17 @@ func (e *Endpoint) initalDiscovery(ctx context.Context) {
...
-		return err
+		switch e.Parent.DisconnectedServersBehavior {
+		case "error":
+			return err
+		case "ignore":
+			// Ignore the error and postpone the init until next collection cycle
+			e.log.Warnf("Error connecting to vCenter on init: %s", err)
+			return nil
+		default:
+			return fmt.Errorf("'%s' is not a valid value for disconnected_servers_behavior",
```
Comment (srebhan) [inline plugins/inputs/vsphere/endpoint.go:315]: Please do not quote the string manually...
```suggestion
			return fmt.Errorf("%q is not a valid value for disconnected_servers_behavior",
```

File: plugins/inputs/vsphere/vsphere_test.go [hunk 2]
```diff
@@ -449,6 +451,22 @@ func TestCollectionNoClusterMetrics(t *testing.T) {
...
+	v.DisconnectedServersBehavior = "error"
+	_, err = NewEndpoint(context.Background(), v, u, v.Log)
+	require.Error(t, err)
+	v.DisconnectedServersBehavior = "ignore"
+	_, err = NewEndpoint(context.Background(), v, u, v.Log)
+	require.NoError(t, err)
+	v.DisconnectedServersBehavior = "something else"
+	_, err = NewEndpoint(context.Background(), v, u, v.Log)
+	require.Error(t, err)
+	require.Equal(t, err.Error(), "'something else' is not a valid value for disconnected_servers_behavior")
```
Comment (srebhan) [inline plugins/inputs/vsphere/vsphere_test.go:467]:
```suggestion
	require.Equal(t, err.Error(), `"something else" is not a valid value for disconnected_servers_behavior`)
```

**Review by srebhan — COMMENTED** [review]
Just have one comment about quoting the string, otherwise looks good. Do you plan to rebase this to get rid of the non-related changes?

File: plugins/inputs/vsphere/README.md [hunk 3]
```diff
@@ -234,6 +234,12 @@ to use them.
   ## The Historical Interval value must match EXACTLY the interval in the daily
   # "Interval Duration" found on the VCenter server under Configure > General > Statistics > Statistic intervals
   # historical_interval = "5m"
+
+  ## Specifies plugin behavior regarding disconnected servers
+  ## Available choices :
+  ##   - error: telegraf will return an error on startup if one the servers is unreachable
+  ##   - skip: telegraf will skip unreachable servers on both startup and gather
```
Comment (yjiabj123) [inline plugins/inputs/vsphere/README.md:241]: Might be 'ignore' instead of 'skip'?

**Pull request merged**